templates/company-board/contracts/CompanyBoardTemplate.sol

```diff
@@ -47,116 +50,120 @@ contract CompanyBoardTemplate is BaseTemplate {
         MiniMeToken shareToken = _createToken(_shareTokenName, _shareTokenSymbol, SHARE_TOKEN_DECIMALS);
-        _storeCache(dao, boardToken, shareToken, msg.sender);
+        _cacheDao(dao, boardToken, shareToken);
     }
 
     /**
-    * @dev Setup a previous deployed DAO with the Company with Board components
+    * @dev Setup a previous deployed DAO with the Board components
+    * @param _members Array of board member addresses (1 token will be minted for each board member)
+    * @param _votingSettings Array of [supportRequired, minAcceptanceQuorum, voteDuration] to set up the board voting app of the organization
+    */
+    function setupBoard(address[] _members, uint64[3] _votingSettings) external {
+        require(_members.length > 0, ERROR_MISSING_BOARD_MEMBERS);
+        require(_votingSettings.length == 3, ERROR_BAD_VOTE_SETTINGS);
+
+        Kernel dao = _fetchDaoCache();
+        _setupBoardApps(dao, _members, _votingSettings);
+    }
+
+    /**
+    * @dev Setup a previous deployed DAO with the Company with Share components
     * @param _id String with the name for org, will assign `[id].aragonid.eth`
-    * @param _boardMembers Array of board member addresses (1 token will be minted for each board member)
-    * @param _shareHolders Array of share holder addresses
-    * @param _shareStakes Array of token stakes for share holders (token has 18 decimals, multiply token amount `* 10^18`)
-    * @param _boardVotingSettings Array of [supportRequired, minAcceptanceQuorum, voteDuration] to set up the board voting app of the organization
-    * @param _shareVotingSettings Array of [supportRequired, minAcceptanceQuorum, voteDuration] to set up the share voting app of the organization
+    * @param _holders Array of share holder addresses
+    * @param _stakes Array of token stakes for share holders (token has 18 decimals, multiply token amount `* 10^18`)
+    * @param _votingSettings Array of [supportRequired, minAcceptanceQuorum, voteDuration] to set up the share voting app of the organization
     * @param _financePeriod Initial duration for accounting periods, it can be set to zero in order to use the default of 30 days.
     * @param _useAgentAsVault Boolean to tell whether to use an Agent app as a more advanced form of Vault app
     */
-    function setupInstance(
-        string _id,
-        address[] _boardMembers,
-        address[] _shareHolders,
-        uint256[] _shareStakes,
-        uint64[3] _boardVotingSettings,
-        uint64[3] _shareVotingSettings,
-        uint64 _financePeriod,
-        bool _useAgentAsVault
-    )
-        external
-    {
-        Kernel dao = _popDaoCache(msg.sender);
-        _setupApps(dao, _boardMembers, _shareHolders, _shareStakes, _boardVotingSettings, _shareVotingSettings, _financePeriod, _useAgentAsVault);
+    function setupShare(string _id, address[] _holders, uint256[] _stakes, uint64[3] _votingSettings, uint64 _financePeriod, bool _useAgentAsVault) external {
```

Comment:
If I understand it correctly now, the flow is:

1. `prepareInstance()`
1. `setupBoard()`
1. `setupShare()`

Is it not possible to combine `prepareInstance()` with `setupBoard()`?

Reply:
It may be possible since they cost 5e6 and 1.3e6 respectively, which means they should fit in a single tx